internal/rbd/nodeserver.go

```diff
@@ -1190,26 +1197,53 @@ func (ns *NodeServer) xfsSupportsReflink() bool {
 func (ns *NodeServer) NodeGetVolumeStats(
 	ctx context.Context,
 	req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	var err error
-	targetPath := req.GetVolumePath()
-	if targetPath == "" {
-		err = fmt.Errorf("targetpath %v is empty", targetPath)
+	var (
+		imgInfo rbdImageMetadataStash
+		err     error
+	)
+	volumePath := req.GetStagingTargetPath()
+	if volumePath == "" {
+		if volumePath = req.GetVolumePath(); volumePath == "" {
+			return nil, status.Error(codes.InvalidArgument, "volume path must be provided")
+		}
+	} else {
+		imgInfo, err = lookupRBDImageMetadataStash(volumePath)
```

Comment:
volumePath will be statingTargetPath+volumeID incase if its a stagingpath https://github.com/ceph/ceph-csi/blob/28fef9b3797e41f277d11d9a3bde11e971994503/internal/rbd/nodeserver.go#L284-L285?